homeassistant/components/withings/common.py

```diff
@@ -1030,6 +1030,21 @@ async def async_get_data_manager(
     if const.DATA_MANAGER not in config_entry_data:
         profile = config_entry.data.get(const.PROFILE)
 
+        if CONF_CLOUDHOOK_URL not in config_entry.data:
+            try:
+                webhook_url = await cloud.async_create_cloudhook(
+                    hass, config_entry.data[CONF_WEBHOOK_ID]
+                )
+            except cloud.CloudNotConnected:
+                webhook_url = webhook.async_generate_url(
+                    hass, config_entry.data[CONF_WEBHOOK_ID]
+                )
+            else:
+                data = {**config_entry.data, CONF_CLOUDHOOK_URL: webhook_url}
+                hass.config_entries.async_update_entry(config_entry, data=data)
+        else:
+            webhook_url = config_entry_data[CONF_CLOUDHOOK_URL]
```

Comment:
```suggestion
            webhook_url = config_entry_data[CONF_CLOUDHOOK_URL]
```

☝️ This shows the code is untested.